lib/axiom/fs/stream/post_message_streams.js

```diff
@@ -0,0 +1,183 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import GenericStreams from 'axiom/fs/stream/generic_streams';
+import EventWithCallback from 'axiom/fs/stream/generic_streams';
+
+/**
+ * Extend a GenericStreams over a messaging port.
+ *
+ * @constructor
+ * @extends {GenericStreams}
+ */
+export var PostMessageStreams = function() {
+  GenericStreams.call(this);
+
+  /** @type {chrome.runtime.Port} */
+  this.port = null;
+  /** @private @type {?Function} */
+  this.onOpenResolve_ = null;
+  /** @private @type {?Function} */
+  this.onOpenReject_ = null;
+  /** @private @type {GenericStreams.ConnectionState} */
+  this.connectionState_ = GenericStreams.ConnectionState.CLOSED;
+
+  // Bind event handlers
+  this.onConnectedMessage_ = this.onConnectedMessage_.bind(this);
+  this.onConnectionError_ = this.onConnectionError_.bind(this);
+  this.onPostMessage_ = this.onPostMessage_.bind(this);
+  this.onDisconnect_ = this.onDisconnect_.bind(this);
+};
+
+PostMessageStreams.prototype = Object.create(GenericStreams.prototype);
+
+export default PostMessageStreams;
+
+/**
+ * @return {Promise}
+ * @param {*} argObject
```

Comment:
This was during experimenting with using a simple object for open so that I could extend upon it with other `open`s, before I realized that there should be separate open methods per subclass.